src/main/java/org/sagebionetworks/bridge/redis/RedisLock.java

```diff
@@ -0,0 +1,57 @@
+package org.sagebionetworks.bridge.redis;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.UUID;
+
+import org.sagebionetworks.bridge.lock.Lock;
+import org.sagebionetworks.bridge.lock.LockNotAvailableException;
+
+public class RedisLock implements Lock {
+
+    private final JedisOps ops;
+
+    public RedisLock(JedisOps ops) {
+        checkNotNull(ops);
+        this.ops = ops;
+    }
+
+    @Override
+    public String acquireLock(final String key, final int expireInSeconds) {
+        checkNotNull(key);
+        checkArgument(expireInSeconds > 0);
+        final String lock = UUID.randomUUID().toString();
+        final Long result = ops.setnx(key, lock);
+        if (result != 1L) {
+            // When fail to obtain the lock
+            // Force to expire the key if not already set to expire
+            final Long expire = ops.ttl(key);
+            if (expire < 0L || expire > expireInSeconds) {
+                expire(key, expireInSeconds);
+            }
```

Comment:
The original logic in RedisLockDao only had the expire < 0L clause (for if the lock had already expired). It looks like this will also expire the lock if another thread has already acquired the lock. Is this something we want to do? Or am I misreading this?